django/db/models/indexes.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import unicode_literals
+
+import hashlib
+
+from django.core.exceptions import ValidationError
+from django.utils.encoding import force_bytes
+
+__all__ = ['Index']
+
+# The max length of the names of the indexes (restricted to 30 due to Oracle)
+MAX_NAME_LENGTH = 30
+
+
+class Index(object):
+
+    index_type = 'idx'
+
+    def __init__(self, *fields, **kwargs):
```

Comment:
> I don't think we should hijack *args for fields, a fields kwarg makes more sense to me.

 Why is that a hijack ? The argument of an `Index` class is it's fields. I used it mainly because defining `Index('foo')` seems syntactically much cleaner than `Index(fields=['foo'])` to me. Am I missing something here ? I agree on replacing `**kwargs` with `name=None` though.

> model should be assigned by Options just like Field.model is.

This is probably more suited once we introduce `indexes` to `Options`, let's keep that for the [next checkpoint](https://github.com/akki/django/pull/1).

> What do you think of the name, db_name approach suggested above? In this case the signature should look like (name, fields, db_name=None, **options) and there would be no need to validate name.

`db_name` still needs to validated if not `name`. I have left some other comments regarding the approach [here](https://github.com/django/django/pull/6726#discussion_r67832769).